source/material_model/visco_plastic.cc

```diff
@@ -171,8 +222,61 @@ namespace aspect
             {
               phase_inputs.phase_index = j;
               phase_function_values[j] = phase_function.compute_value(phase_inputs);
+
+            }
+
+          if ((enable_material_lookup) & this->simulator_is_past_initialization() & (out.additional_outputs.size()>1))
+            {
+              const std::vector<unsigned int> n_phases_per_composition = phase_function_lookup.n_phases_for_each_composition();
+              unsigned int n_phase_transition_cur = 0;
+              unsigned int n_phase_cur = 0;
+              for (unsigned int c=0; c<(in.composition[i].size()+1); ++c)
+                {
+                  const std::vector<std::string> phase_names = equation_of_state_lookup.get_phase_names(c);
+                  unsigned int index_of_lookup_phase;
+
+                  std::vector<std::string> transition_lookup_names(transition_lookup_phases.begin() + n_phase_cur, transition_lookup_phases.begin()+ n_phase_cur + n_phases_per_composition[c]);
+                  const MaterialModel::NamedAdditionalMaterialOutputs<dim> *result
+                    = dynamic_cast<const MaterialModel::NamedAdditionalMaterialOutputs<dim> *> (out.additional_outputs[2].get());
+
+                  if (result)
+                    {
+                      std::vector<double> dominant_phase_index_vals = result->get_nth_output(0); //
+                      std::string name_dominant_phase = phase_names[static_cast<int>(dominant_phase_index_vals[i])];
+                      std::vector<std::string>::iterator it = std::find(transition_lookup_names.begin(),
+                                                                        transition_lookup_names.end(),
+                                                                        name_dominant_phase);
+
+                      if (it != transition_lookup_names.end())
+                        {
+                          index_of_lookup_phase = std::distance(transition_lookup_names.begin(), it);
+                        }
+                      else
```

Comment:
A lot of this logic could be accomplished in the parameter parsing phase. I put it there and now have things set up, so that each phase that can be a dominant phase has it's own place in the input .prm file.